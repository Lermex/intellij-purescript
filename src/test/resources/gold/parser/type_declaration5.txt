Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSTypeSynonymDeclarationImpl(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PsiElement(proper name)('Foo')
          PsiElement(whitespace)(' ')
        PSGenericIdentifierImpl(GenericIdentifier)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PsiElement(()('(')
            PSTypeImpl(Type)
              PSTypeAtomImpl(TypeAtom)
                PSTypeConstructorImpl(TypeConstructor)
                  PSQualifiedImpl(Qualified)
                    PSProperNameImpl(ProperName)
                      PsiElement(proper name)('Array')
                      PsiElement(whitespace)(' ')
              PSTypeAtomImpl(TypeAtom)
                PSTypeConstructorImpl(TypeConstructor)
                  PSQualifiedImpl(Qualified)
                    PSProperNameImpl(ProperName)
                      PsiElement(proper name)('User')
            PsiElement())(')')
            PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSTypeImpl(Type)
            PSTypeAtomImpl(TypeAtom)
              PSGenericIdentifierImpl(GenericIdentifier)
                PsiElement(identifier)('a')