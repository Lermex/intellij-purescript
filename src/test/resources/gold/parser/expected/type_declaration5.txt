Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Main')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSTypeDeclarationImpl(TYPE_DECLARATION)
        PsiElement(PSTokenType.type)('type')
        PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('Foo')
        PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
          PSDeclarationRefIdImpl(DECLARATION_REF_ID)
            PsiElement(PSTokenType.ID)('a')
        PsiElement(PSTokenType.=)('=')
        PsiElement(PSTokenType.()('(')
        PSTypeDeclarationItemImpl(TYPE_DECLARATION_ITEM)
          PSTypeDeclarationTypeOnlyImpl(TYPE_DECLARATION_TYPE_ONLY)
            PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
              PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('Array')
                PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                  PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('User')
                  PsiElement(PSTokenType.))(')')
            PsiElement(PSTokenType.->)('->')
            PSParenthesisedTypeOrIdListImpl(PARENTHESISED_TYPE_OR_ID_LIST)
              PSParenthesisedDeclarationRefIdListImpl(PARENTHESISED_DECLARATION_REF_ID_LIST)
                PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                  PsiElement(PSTokenType.ID)('a')